### ISSUE TITLE
Can the NavigationID for a NavigationComplete appear twice?

### ISSUE THREAD

**Issue by darbid** [MicrosoftEdge/WebView2Feedback#1142]
I was tracking Navigation Starting and Complete events and noticed that after navigating from a Google search result a NavigationComplete event with the same NavigationID fires again.

I have created a [Repo example here](https://github.com/darbid/WebView2NavigationEventsTest) using  Version="1.0.774.44"

1. The repo opens with WebView2's source being set to a Google search.
2. Click on the first Microsoft result.

The output will look like this; 

![image](https://user-images.githubusercontent.com/3191525/113377899-e6594980-9375-11eb-8639-d64f77d0263a.png)

My question is why does WebView2 have a NavigationComplete with ID 2 after you click on the Microsoft link?

Also I must say that i have not picked up on this anywhere else it is just the combination of Google search and then clicking on a result.

[AB#32473444](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/32473444)

**Comment by champnic** [comment]
Hey @darbid - we just gave your repro a try and didn't see this behavior:
```
#######################NavigationStarting##############################
NavigationId: 2
URL: https://www.google.com/search?q=webview2
#####################################################
++++++++++++++++++++++++NavigationCompleted++++++++++++++++++++++++++++
NavigationId: 2
URL: https://www.google.com/search?q=webview2
++++++++++++++++++++++++++++++++++++++++++++++++++++
#######################NavigationStarting##############################
NavigationId: 6
URL: https://docs.microsoft.com/en-us/microsoft-edge/webview2/
#####################################################
++++++++++++++++++++++++NavigationCompleted++++++++++++++++++++++++++++
NavigationId: 6
URL: https://docs.microsoft.com/en-us/microsoft-edge/webview2/
++++++++++++++++++++++++++++++++++++++++++++++++++++
The thread 0xc3e8 has exited with code 0 (0x0).
The thread 0x476c has exited with code 0 (0x0).
```

What runtime version are you using? We tested with 89.0.774 stable runtime.

**Comment by darbid** [comment]
Sorry for not adding this information.

I have just tried it with 1.0.705.50 and setting 

` <WV:WebView2.CreationProperties>
                <WV:CoreWebView2CreationProperties BrowserExecutableFolder="C:\Program Files (x86)\Microsoft\Edge\Application\89.0.774.75" />
            </WV:WebView2.CreationProperties>`

OR

`<WV:WebView2.CreationProperties>
                <WV:CoreWebView2CreationProperties BrowserExecutableFolder="C:\Program Files (x86)\Microsoft\Edge Dev\Application\91.0.845.2" />
            </WV:WebView2.CreationProperties>`

I repro it with 1.0.705.50 and 89.0.774.75
I do not repro it with 1.0.705.50 and 91.0.845.2

I repro it with 1.0.774.44 and 91.0.845.2
I repro it with 1.0.824 and 91.0.845.2

**Comment by champnic** [comment]
Thanks for the added info. I don't think this is By Design so I've opened a bug on our backlog to take a closer look.

**Comment by billxc** [comment]
Hey @darbid - we have a fix for this in runtimes 93.0.950.0+. You can use Edge Canary to try this out.

**Comment by darbid** [comment]
Thanks for the detailed feedback. Very interesting to know what it is. And thanks for pointing out a coding weekness, always good to get help.
